Compute Inertial Strahlkorper if Strahlkorper frame is not inertial.

markscheel opened the pull request:
Now we can observe quantities in the Inertial frame, even if the
Strahlkorper is found in a different frame.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by
explaining why you chose the solution you did and what alternatives you
considered, etc...
-->

kidder commented on src/NumericalAlgorithms/Interpolation/InterpolationTargetApparentHorizon.hpp:
```diff
@@ -141,19 +141,30 @@ template <typename InterpolationTargetTag, typename Frame>
-                              logging::Tags::Verbosity<InterpolationTargetTag>,
-                              ::ah::Tags::PreviousStrahlkorper<Frame>>,
-                   StrahlkorperTags::compute_items_tags<Frame>>;
+  using initialization_tags = tmpl::append<
```
This type alias should not exist.  I could not understand how this could possibly work since `initialization_tags` should only be simple tags.  It turns out that as far as I can tell it is only used to fill the unused type alias `return_tag_list` and in some detail metafunctions that can now be eliminated in favor of using CreateGetTypeAliasOrDefault and if-constexpr  in `InitializeInterpolationTarget.hpp`, and (It is also present and should be removed form the KerrHorizon target.).

kidder commented on src/NumericalAlgorithms/Interpolation/InterpolationTargetApparentHorizon.hpp:
```diff
@@ -167,9 +178,18 @@ struct ApparentHorizon {
-        options.initial_guess);
+    if constexpr (std::is_same_v<Frame, ::Frame::Inertial>) {
+      Initialization::mutate_assign<simple_tags>(
+          box, options.initial_guess, options.fast_flow, options.verbosity,
```
this depends upon `StrahlkoperTags::items_tags<Frame>` being exactly `tmpl::list<Strahlkorper<Frame>>` so why not just use that in the `simple_tags` list above?

kidder commented on src/NumericalAlgorithms/Interpolation/InterpolationTargetApparentHorizon.hpp:
```diff
@@ -141,19 +141,30 @@ template <typename InterpolationTargetTag, typename Frame>
   using frame = Frame;
 
-  using simple_tags =
+  using simple_tags = std::conditional_t<
```
I thought we preferred tmpl::conditional_t over std::conditional_t

kidder commented on src/NumericalAlgorithms/Interpolation/InterpolationTargetApparentHorizon.hpp:
```diff
@@ -167,9 +178,18 @@ struct ApparentHorizon {
+      // ... and also default-initialize Strahlkorper in inertial
+      // frame, if we are finding the Strahlkorper in a non-inertial
+      // frame.
+      Initialization::mutate_assign<simple_tags>(
```
all simple tags are already default initialized so you could just mutate_assign<common_tags> and get rid of this if-constexpr

kidder commented on src/NumericalAlgorithms/Interpolation/InterpolationTargetApparentHorizon.hpp:
```diff
@@ -167,9 +178,18 @@ struct ApparentHorizon {
+    }
   }
 
   template <typename Metavariables, typename DbTags, typename TemporalId>
```
shouldn't there be something added to a test somewhere to test the changes made in this PR?

kidder commented on src/NumericalAlgorithms/Interpolation/InterpolationTargetApparentHorizon.hpp:
```diff
@@ -141,19 +141,30 @@ template <typename InterpolationTargetTag, typename Frame>
       tmpl::push_back<StrahlkorperTags::items_tags<Frame>, ::ah::Tags::FastFlow,
                       logging::Tags::Verbosity<InterpolationTargetTag>,
-                      ::ah::Tags::PreviousStrahlkorper<Frame>>;
+                      ::ah::Tags::PreviousStrahlkorper<Frame>>,
```
how about making a type alias `common_tags` so they don't get out of sync in the future.   Then this can be `using simple_tags = tmpl::append<common_tags, tmpl::conditional_t<...>>;`

markscheel commented:
I will rebase and drastically modify this PR once #3524 is merged.

markscheel committed:
Always compute Inertial Strahlkorper.

...even if the Strahlkorper is found in a frame that is not
the Inertial frame. This is because we often want to observe
in the Inertial frame.

markscheel commented:
Redid this PR, including adding a test.
